source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -81,6 +81,14 @@ void CheckRequestUtils::setAttrContextPeer(envoy::service::auth::v3::AttributeCo
   }
 }
 
+std::string CheckRequestUtils::getScheme(const Envoy::Http::RequestHeaderMap& headers) {
+  const auto* scheme_xfp = headers.ForwardedProto();
+  if (scheme_xfp == nullptr || scheme_xfp->value().empty()) {
+    return getHeaderStr(headers.Scheme());
+  }
+  return std::string(scheme_xfp->value().getStringView());
```

Comment:
so the behavior we want is for `x-forwarded-proto` to take precendence. Given the behavior up until now has been that `:scheme` takes presesedence do we see this as a "breaking behavior" change?

Reply:
Hum, you're right. While this is a "metadata" (completeness) only (I'm not sure someone would take a decision of authenticating request based on the scheme), do you think to document this is enough?

Reply:
I'm not too familiar with expectations for this filter, so can't say one way or another. cc @gsagula so that both of you as codeowners can decide.